lib/fog/aliyun/models/compute/security_group_rule.rb

```diff
@@ -19,22 +19,28 @@ class SecurityGroupRule < Fog::Model
         attribute :priority,                  :aliases => 'Priority'
 
         def save
-          requires :ip_protocol, :from_port, :to_port, :parent_group_id
-          cidr = ip_range && ip_range["cidr"]
-          if rule = service.create_security_group_rule(parent_group_id, ip_protocol, from_port, to_port, cidr, group).data[:body]
-            merge_attributes(rule["security_group_rule"])
+          requires :security_group_id,:ip_protocol,:port_range,:direction
+          options={}
+          options[:portRange] = port_range if port_range
+          options[:policy] = policy if policy
+          options[:priority] = priority if port_range
+          options[:protocol] = ip_protocol if ip_protocol
+          if direction=="egress"
+            if dest_cidr_ip
+              service.create_security_group_egress_ip_rule(security_group_id,dest_cidr_ip,nic_type,options)
+            elsif dest_group_id
+              options[:destGroupOwnerAccount] = dest_owner if dest_owner
+              service.create_security_group_egress_sg_rule(security_group_id,dest_group_id,options)
+            end
+          else
+            if source_cidr_ip
+              service.create_security_group_ip_rule(security_group_id,source_cidr_ip,nic_type,options)
+            elsif source_group_id
+              options[:sourceGroupOwnerAccount] = source_owner if source_owner
+              service.create_security_group_sg_rule(security_group_id,source_group_id,options)
+            end
           end
         end
-
-        def destroy
-          requires :id
-          service.delete_security_group_rule(id)
-          true
-        end
```

Comment:
Maybe you could leave the `destroy` method to maintain the same interface as the majority of `fog` but calling the service `revoke_seurity_group_ig_rule` instead. What do you think?